tensorflow/core/tpu/tpu_initializer_helper.cc

```diff
@@ -51,6 +54,62 @@ bool GetEnvBool(const char* name, bool defval) {
 
 }  // namespace
 
+bool PrintFdMessage(long tgid) {
+  char path[40], line[100];
+  DIR* fd;
+  struct dirent* ent;
+  long deviceId;
+  snprintf(path, 40, "/proc/%ld/fd", tgid); 
```

Comment:
Use `absl::StrCat("/proc/", tgid, "/fd")`, that should simplify this a bunch (here and below)

Reply:
The problem with it is that it returns string, and opendir and readlink require const char *, so by using this we need to convert it back to char [] by using .c_str() function. I did it but I am not sure if it is better than just using char from the beginning. Let me know what you think.

Reply:
Ah, I incorrectly though data() had a non-const overload. Maybe just use `&line[0]` then: https://stackoverflow.com/questions/1919626/can-i-get-a-non-const-c-string-back-from-a-c-string

Reply:
(Oops, not sure how I posted that twice)